### DISCUSSION TITLE
Is it possible to access the row virtualizer instance?

### DISCUSSION THREAD

**Discussion by mihaio100** [KevinVandy/material-react-table#45] [Q&A] [answered]
I have a table with `virtualization` enabled and I've been trying to figure out how to use the `scrollToIndex` function provided by react-virtual (as per [docs](https://tanstack.com/virtual/v3/docs/api/virtualizer?from=reactVirtualV2&original=https://react-virtual-v2.tanstack.com/docs/api#scrolltoindex)). However I can't seem to figure out how to get the `Virtualizer` instance in order to access that function, or whether this is even possible at all at the moment? 

Looking through the source code, it looks like a `rowVirtualizer` instance is created inside `MRT_TableBody` but isn't being exposed in any way, so would there be a way for me to use `scrollToIndex` through MRTable?

**Comment by KevinVandy** [comment, marked as answer]
Edit: There is now a `virutalizerInstanceRef` prop. It can be seen used in the [infinite scrolling example](https://www.material-react-table.com/docs/examples/infinite-scrolling)

So there is currently no reference to the virtualizer currently, though I could probably expose a ref to it.

If all you are wanting to do is scroll to the top, you could do that with the `muiTableContainerProps` ref.

Something like this

```ts
  const tableContainerRef = useRef(null);

  //scroll to top of table when sorting or filters change
  useEffect(() => {
    if (tableContainerRef.current) {
      tableContainerRef.current.scrollTop = 0;
    }
  }, [sorting, columnFilters, globalFilter]);

...
<MaterialReactTable
  muiTableContainerProps={{
    ref: tableContainerRef
  }}
/>
```

But adding a virtualizerRef to the library itself is probably a good idea and not too dangerous. I'll see if that works and maybe add it soon.

**Reply by mihaio100** [reply]
Thank you for clarifying!

**Reply by KevinVandy** [reply]
So actually, check out the latest version and the infinite [scrolling example](https://www.material-react-table.com/docs/examples/infinite-scrolling) on the docs site. I added a virtualizerInstanceRef prop

**Reply by LukeParsnips** [reply]
Thanks so much for this! We were just getting ready to Fork & create a PR so we appreciate the < 24h turnaround.
